include/particles/particle.tcc

```diff
@@ -500,6 +500,25 @@ void mpm::Particle<Tdim>::map_mass_momentum_to_nodes() noexcept {
   }
 }
 
+//! Map nodal properties to nodes
+template <unsigned Tdim>
+void mpm::Particle<Tdim>::map_properties_to_nodes() noexcept {
+  // Check if particle mass is set
+  assert(mass_ != std::numeric_limits<double>::max());
+
+  // Unit 1x1 Eigen matrix to be used with scalar quantities
+  Eigen::Matrix<double, 1, 1> unit_scalar;
+  unit_scalar << 1.0;
+
+  // Map mass and momentum to nodal property taking into account the material id
+  for (unsigned i = 0; i < nodes_.size(); ++i) {
+    nodes_[i]->update_property(
+        true, "masses", mass_ * shapefn_[i] * unit_scalar, material_id_, 1);
```

Comment:
I don't follow the multiplication with a scalar, why are we doing that?
```suggestion
        true, "masses", mass_ * shapefn_[i] , material_id_, 1);
```

Reply:
Because `mass_` is not an Eigen matrix 1 x 1 (neither is shapefn_[i]). Therefore, the update_property function will fail because we are not passing an argument of the correct type.